cfe_internal/update/update_policy.cf

```diff
@@ -80,6 +80,47 @@ bundle agent cfe_internal_update_policy
 
 }
 
+bundle agent cfe_internal_remove_broken_symlink(symlink_path)
+# @brief Checks if the symlink_path symlink points to an existing target
+# @warning Only works for symlinks using absolute paths!
+{
+  vars:
+      "link_target" string => filestat("$(symlink_path)", "linktarget");
+
+  files:
+      "$(symlink_path)"
+        delete => u_tidy,
+        if => not(fileexists("$(link_target)"));
```

Comment:
Since there is a promise later to make sure the symlink exists, I think we don't need this separate bundle to remove a potentially broken symlink first. We can instead use move_obstructions => "true" to ensure the symlink target is changed when a broken one exists.

Reply:
So you want the symlink to be recreated with every single `update.cf` policy run? Or am I missing something? I don't see how [`move_obstructions`](https://docs.cfengine.com/docs/3.12/reference-promise-types-files.html#move_obstructions) would only replace a broken symlink.

Reply:
move_obstructions as I understand, will move the broken symlink out of the way, or if someone had created a directory at symlink_path, it would move the directory, or if someone had created a plain file at symlink_path, it would remove the file and replace it with the symlink. move_obstructions should not cause the symlink to be re-created each agent run, it just allows the promise to override a mismatch in file types.

Reply:
OK, I'll test it on some small example and see what it does.

Reply:
See my comment below for a small example